javascript/compute-height.js

```diff
@@ -0,0 +1,29 @@
+const TAB_OFFSET_PADDING = 5;
+
+function getRecursiveParentNode(el, n) {
+    if(n <= 0) return el;
+    if(el === null || el.parentNode === null) return el;
+    return getRecursiveParentNode(el.parentNode, n-1);
+}
+
+const getDynamicElementFromContainer = (container) => {
+    const webuiParentDepth = 7;
+    return getRecursiveParentNode(container, webuiParentDepth);
+}
+
+function computeComfyuiElementHeight() {
+    const container = document.getElementById("comfyui_webui_container");
+    const dynamicElement = getDynamicElementFromContainer(container);
+
+    if(dynamicElement !== null) {

```

Comment:
I'd invert the polling condition:

```suggestion
    if(dynamicElement === null) {
        setTimeout(computeComfyuiElementHeight, 200);
        return;
    }
    // rest of function as normal, un-nested code...
```

Reply:
and then call setTimeout twice?
```

    if(dynamicElement === null) {
        // polling ew
        setTimeout(computeComfyuiElementHeight, 200); // 1
        return;
    }

    const height = dynamicElement.offsetHeight;
    container.style.height = `calc(100% - ${height-TAB_OFFSET_PADDING}px)`;
    setTimeout(computeComfyuiElementHeight, 200); // 2
```

Reply:
else, we can do this:
```js
    // polling ew
    setTimeout(computeComfyuiElementHeight, 200);

    if(dynamicElement === null) {
        return;
    }

    const height = dynamicElement.offsetHeight;
    container.style.height = `calc(100% - ${height-TAB_OFFSET_PADDING}px)`;
```

but I'd like for the setTimeout to be at the end for clarity

Reply:
My issue is with the nesting of the function code, but I guess in this case it's not really possible without duplicating the weird return logic with setTimeout. I guess you can pick your poison.